Bottom: 5ec1da356d5808f2fa755ce562b04eaf2f7eb750
Top:    b053011fc79ad5b43c47ad140e990e7f99a120de
Author: Jason Wang <jasowang@redhat.com>
Date:   2016-06-15 13:37:48 +0800

Refresh of tun-introduce-tx-array-mode

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 20bebc7..b1f1612 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1569,18 +1569,15 @@ static ssize_t tun_do_read(struct tun_struct *tun, struct tun_file *tfile,
 	if (!iov_iter_count(to))
 		return 0;
 
-	if (tun->flags & IFF_TX_ARRAY) {
+	if (tun->flags & IFF_TX_ARRAY)
 		skb = tun_ring_recv(tfile, noblock, &err);
-		if (!skb)
-			return err;
-	} else {
+	else
 		/* Read frames from queue */
 		skb = __skb_recv_datagram(tfile->socket.sk,
 					  noblock ? MSG_DONTWAIT : 0,
 					  &peeked, &off, &err);
-		if (!skb)
-			return err;
-	}
+	if (!skb)
+		return err;
 
 	ret = tun_put_user(tun, tfile, skb, to);
 	if (unlikely(ret < 0))
